### ISSUE TITLE
Web Resource page won't open after upgrading to GLPI 10

### ISSUE THREAD

**Issue by hgpit** [cconard96/glpi-webresources-plugin#31]
**Describe the bug**
When browsing to the Web Resources landing page (ie: by navigating to it via the GLPI menu), the page just endlessly loads:

![image](https://user-images.githubusercontent.com/16598973/171991258-0ef19f45-e031-44de-a72e-1902b5d4f52b.png)

**To Reproduce**
Browse to here:
![image](https://user-images.githubusercontent.com/16598973/171991278-8cd84bd7-5854-44a5-84c7-0e9fb82b5c80.png)

**Expected behavior**
Directory of all our web resources entries should appear.

**System Specifications (please complete the following information):**
 - GLPI version: 10.0.1
 - GLPI Server OS: CentOS 8.5 (4.18.0-348.12.2.el8_5.x86_64)
 - Browser: Microsoft Edge 102.0.1245.30 (Official build) (64-bit)
 - Plugin version: 2.0.1

**Additional context**
This is what our Apache log records:
```
[Sat Jun 04 09:21:15.729179 2022] [proxy_fcgi:error] [pid 1930066:tid 140538669901568] [client 10.11.252.150:4204] malformed header from script 'refreshDashboard.php': Bad header: , referer: https://helpdesk.hayley-group.local/marketplace/webresources/front/dashboard.php
[Sat Jun 04 09:21:15.729341 2022] [proxy_fcgi:error] [pid 1930066:tid 140538669901568] [client 10.11.252.150:4204] AH01070: Error parsing script headers, referer: https://helpdesk.hayley-group.local/marketplace/webresources/front/dashboard.php
[Sat Jun 04 09:21:15.729354 2022] [proxy_fcgi:error] [pid 1930066:tid 140538669901568] (22)Invalid argument: [client 10.11.252.150:4204] AH01075: Error dispatching request to : , referer: https://helpdesk.hayley-group.local/marketplace/webresources/front/dashboard.php
```

**Comment by hgpit** [comment]
Browser JS console reports this:

![image](https://user-images.githubusercontent.com/16598973/171992357-f1eec556-4190-48e4-94d5-7019f96595ce.png)

**Comment by cconard96** [comment]
It sounds like a PHP error is occurring. Please check your GLPI logs. Specifically, `files/_log/php-errors.log`.

**Comment by hgpit** [comment]
Ooh, good call. I was really hoping that would lead me to the fix, but disappointingly, php-errors.log isn't logging anything when attempting to reach this page (I tried it a few times, even smashed CTRL-F5).

The last thing logged was (which I don't believe is directly related, but interestingly does refer to header issues again):

```
[root@g000077 _log]# tail -f php-errors.log
  ...r/laminas/laminas-mail/src/Storage/Pop3.php:105 Laminas\Mime\Decode::splitMessage()
  ...r/laminas/laminas-mail/src/Storage/Part.php:147 Laminas\Mail\Storage\Pop3->getRawContent()
  src/MailCollector.php:2277                         Laminas\Mail\Storage\Part->getContent()
  src/MailCollector.php:1741                         MailCollector->getDecodedContent()
  src/MailCollector.php:1086                         MailCollector->getBody()
  src/MailCollector.php:761                          MailCollector->buildTicket()
  src/MailCollector.php:1830                         MailCollector->collect()
  src/CronTask.php:1018                              MailCollector::cronMailgate()
  front/cron.php:84                                  CronTask::launch()
  
" does not match header format! in /var/www/helpdesk.hayley-group.local/vendor/laminas/laminas-mail/src/Headers.php at line 116
  Backtrace :
  vendor/laminas/laminas-mime/src/Decode.php:162     Laminas\Mail\Headers::fromString()
  ...r/laminas/laminas-mail/src/Storage/Pop3.php:105 Laminas\Mime\Decode::splitMessage()
  ...r/laminas/laminas-mail/src/Storage/Part.php:147 Laminas\Mail\Storage\Pop3->getRawContent()
  src/MailCollector.php:2277                         Laminas\Mail\Storage\Part->getContent()
  src/MailCollector.php:1741                         MailCollector->getDecodedContent()
  src/MailCollector.php:1086                         MailCollector->getBody()
  src/MailCollector.php:761                          MailCollector->buildTicket()
  src/MailCollector.php:1830                         MailCollector->collect()
  src/CronTask.php:1018                              MailCollector::cronMailgate()
  front/cron.php:84                                  CronTask::launch()
```

**Comment by hgpit** [comment]
Tried disabling and re-enabling plugin.
Tried with all other plugins disabled.
Tried with opcache disabled.

Problem persists.

**Comment by cconard96** [comment]
The header issues with the mail collector is different (many many issues opened on the core GLPI repository about it since the new mail library is overly sensitive about them...).

It is difficult for me to determine what header it is having issues with since it logged an empty value:
```
malformed header from script 'refreshDashboard.php': Bad header: , referer: ...
```

The only header changed/sent specifically by the plugin here is `Content-Type` which looks correct.
There are also headers sent to prevent the browser from caching the content, but that exists on other pages in GLPI itself.

**Comment by hgpit** [comment]
Hmm, damn! I'm going to have to create a separate manual page which draws from the GLPI DB plugin table for now then.
I honestly don't know PHP/GLPI well enough to unravel this issue.

Thanks for your input, though. Hopefully others will encounter the issue and can shed some more light on it.

It might be worth nothing that the other Web Resources plugin areas appear to work okay, such as the list/editing area:
![image](https://user-images.githubusercontent.com/16598973/173318434-105c2bb5-738a-4e4b-863e-076a18e62d77.png)
The problem only seems to appear in the main view page.

**Comment by cconard96** [comment]
The warning is because the plugin still uses the old UI. When it gets moved into Twig templates this deprecation notice will go away. It is something I want to do for a minor version before GLPI 10.1 is released.

**Comment by cconard96** [comment]
Nevermind, I'm clearly dumb. The header being set was invalid. The browser was fixing my mistake automatically but some web servers are less forgiving. Should be fixed by e0777206ca4908bbb541a35b3e03e0dab9d54054.

A new release is expected this morning.
